### ISSUE TITLE
Support custom domains for Request-Driven Web Services

### ISSUE THREAD

**Issue by efekarakus** [aws/copilot-cli#2411]
## Ask
As a customer that initialized an application with a domain and a request-driven web service,  
I want to specify a custom domain name in the manifest file similar to #1188,  
So that I can share a friendly URL for my service instead of the default generated one from App Runner.

### User Experience
We propose to follow a very similar UX to #1188, where we can add the `alias` field to support particular subdomains under the application.
```yaml
http:
  alias: 'api.example.com'
```
After running `copilot deploy`, then the progress tracker should display an additional step:
```
Associate your custom domain with the App Runner subdomain URL   [in progress]
```

### Programming Model

We'll have a custom resource backed by Lambda that will call [AssociateCustomDomain](https://docs.aws.amazon.com/apprunner/latest/api/API_AssociateCustomDomain.html) using the value provided by `alias`.

Then, Copilot should write the CNAME records provided by App Runner to the root hosted zone.

**Comment by efekarakus** [comment]
My intuition is that for test environments people will be happy with the generated domain name from App Runner. Therefore, we only need to provide the `alias` field for prod environments and limiting it to strictly `example.com` or 1-level subdomain `api.example.com` should be good enough:
```yaml
environments:
  prod:
     http:
       alias: 'api.example.com'
```

**Comment by Lou1415926** [comment]
The feature is now released in [v1.9.0](https://github.com/aws/copilot-cli/releases/tag/v1.9.0)! 🎊
